tests/functionaltests/FunctionalTestHelpers.mm

```diff
@@ -30,11 +30,7 @@
 
 // Launches the functional test app
 bool FunctionalTestSetupUIApplication() {
-    RunSynchronouslyOnMainThread(^{
```

Comment:
>RunSynchronouslyOnMainThread [](start = 4, length = 28)

Maybe we should be using CoreDispatcher APIs rather than NSThread, etc for this?  How can you be positive that FunctionalTestSetupUIApplication is getting called from the UI thread?